internal/setup/setup_test.go

```diff
@@ -851,3 +852,81 @@ func (s *S) TestParseRelease(c *C) {
 		}
 	}
 }
+
+// Test YAML marshalling of setup.Package and setup.Slice
+func (s *S) TestPackageMarshalYAML(c *C) {
```

Comment:
I dont fully understand this test, what is it trying to do? I would assume the way to test marshalling would be to take a release object and assert that the result is the expected YAML, right?

Reply:
This test is similar to what you said.
* It parses the `sampleRelease` YAMLs in `release`.
* Converts the `release` packages to YAMLs, writes them on disk and parses again in `newRelease`.
* Checks that `release` and `newRelease` are indeed the same.

It relies on the assumption that parsing is working well (which we are testing in the above function) and instead of comparing YAMLs, we can parse them and then compare the objects. Let me know if you think I should change it.

Reply:
I get it now! I like the idea, the only thing that I think could be improved is reusing the tests cases above instead of creating a new one. Basically either iterate over the tests above in this function and perform the same file -> parse -> marshall ->parse -> check or move the logic to the test function above that already sets up the directories and everything.

Reply:
If the logic is too different and I it looks like that, having a separate test that uses `setupTest` would be the best way forward to not clutter the other one.

Reply:
I have thought about it, I think the best thing would be separate function and iterate over `setupTest` then do:
1. write(test.release) to disk (we are already testing the YAML parsing).
2. read from disk the release that we have just written.
3. compare test.release and the parsed one (we have to do that instead of comparing YAML because the one we are producing is not exactly the same).

Reply:
Added in https://github.com/canonical/chisel/pull/101/commits/e3c2a30e6b8d2622690493ee73f5f10c89dbb403.